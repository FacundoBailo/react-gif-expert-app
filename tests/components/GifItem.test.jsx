import { screen, render } from '@testing-library/react';
import GifItem from '../../src/components/GifItem';

describe('Testando componente GifItem', () => { 

    const title = 'Dragon Ball Z';
    const url = 'https://dragonballz.com/dbz.jpg';

    test('deberia de hacer match con el snapshot', () => { 

      const { container } = render(<GifItem title={title } url={ url }/>);

      expect( container ).toMatchSnapshot();

     })

    test('debe mostrar la imagen con el URL y el ALT indicado', () => { 
 
      render(<GifItem title={title } url={ url }/>);

      const { src, alt } = screen.getByRole('img');
 
      expect( src ).toBe( url );
      expect( alt ).toBe( alt );
 
    }) 

    test('debe mostrar el titulo en el componente', () => { 

      render(<GifItem title={title } url={ url }/>);
      expect(screen.getByText( title )).toBeTruthy();

    })
     
 })