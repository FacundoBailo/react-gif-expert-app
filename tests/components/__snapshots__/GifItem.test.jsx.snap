// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando componente GifItem deberia de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball Z"
      src="https://dragonballz.com/dbz.jpg"
    />
    <p>
      Dragon Ball Z
    </p>
  </div>
</div>
`;
